ui/src/lib/utils/throttle.util.ts

```diff
@@ -0,0 +1,27 @@
+export const throttle = <T extends (...args: any[]) => void>(fn: T, wait: number) => {
+  let lastTime: number
+  let inThrottle: boolean
+  let lastFn: ReturnType<typeof setTimeout>
+
+  return function func(this: any, ...args: any[]): void {
```

Comment:
I think a better implementation would be as such:
1) if we don't have a pending timeout - call the function immediately and start a timeout
2) if we have a timeout - set a flag that the timeout should call the function
3) after the timeout is reached - call the function if the flag is set